pkg/obfuscate/appsec.go

```diff
@@ -0,0 +1,538 @@
+		case scanObjectKey:
+			// Object key scanned and keyFrom and keyTo were set according to
+			// the previous scanBeginLiteral and scanContinue scanner operations
+			if input[keyFrom:keyTo+1] == `"parameters"` {
+				i = o.obfuscateRuleMatchParameters(&scanner, input, i+1, &diff)
+				i-- // decrement i due to the for loop increment but i is already set to the next byte to scan
+			}
+			keyFrom = -1
+			keyTo = -1
+		}
+	}
+	if scanner.eof() == scanError {
+		return input, scanner.err
+	}
+	return diff.apply(input), nil
+}
+
+// Obfuscate the array of parameters of the form `[ <parameter 1>, <...>, <parameter N> ]`.
+// The implementation accepts elements of unexpected types.
+func (o *appsecEventsObfuscator) obfuscateRuleMatchParameters(scanner *scanner, input string, i int, diff *inputDiff) int {
+	i, err := stepTo(scanner, input, i, scanBeginArray)
+	if err != nil {
+		return i
+	}
+	for ; i < len(input); i++ {
+		i, err = o.obfuscateRuleMatchParameter(scanner, input, i, diff)
+		if err != nil {
+			got, ok := err.(unexpectedScannerOpError)
+			if !ok {
+				return i // Abort due to an unexpected error (syntax error or end of json)
+			}
+			switch got {
+			case scanEndArray:
+				// The previous call failed because we reached the end of the array
+				// This case happens for the empty array value
+				return i
+			case scanBeginObject:
+				// Try to step until the next object value
+				i, err = stepUntil(scanner, input, i, scanEndObject)
+			case scanBeginArray:
+				// Try to step until the end of the array value
+				i, err = stepUntil(scanner, input, i, scanEndArray)
+			case scanBeginLiteral:
+				// Let the following stepToOneOf do the job and scan until the next array value
+				err = nil
+			}
+			if err != nil {
+				return i
+			}
+		}
+		// Step to the beginning of the next array value or end of the array
+		var op int
+		i, op, err = stepToOneOf(scanner, input, i, scanArrayValue, scanEndArray)
+		if err != nil || op == scanEndArray {
+			return i
+		}
+	}
+	return i
+}
+
+// Obfuscate the parameter object of the form `{ "key_path": <key path>, "highlight": <highlight>, "value": <value>, <any extra keys>: <any extra values> }`.
+// The implementation is permissive so that any extra keys and values are allowed, and permits having any of the three
+// keys `key_path`, `highlight` and `value` that we need to obfuscate.
+// Note that the overall parameter obfuscation directly depends on the presence of a sensitive key in the key_path.
+// As a result, the parameter object needs to be entirely walked to firstly find the key_path.
+func (o *appsecEventsObfuscator) obfuscateRuleMatchParameter(scanner *scanner, input string, i int, diff *inputDiff) (int, error) {
+	// Walk the object and save the `key_path` value along with the offsets of the
+	// `highlight` and `value` values, if any.
+	var (
+		paramKeyPath                         string
+		paramValueFrom, paramValueTo         int
+		paramHighlightFrom, paramHighlightTo int
```

Comment:
```suggestion
		key        string // key specifies the ...
		pfrom, pto int // pfrom & pto specify the start and end offsets of ...
		hfrom, hto int // hfrom & hto specify the start and end offsets of ...
```
Please try to use the same coding style as the rest of the code with short variable names if they are only used locally. Make reading the algorithm more of a priority than the brevity of the names. The names only need to be read once, understood and memorised. If they are long, it takes too much time to read them in the code, obstructing the actual algorithm logic and making it hard to follow.

Reply:
Fixed by https://github.com/DataDog/datadog-agent/pull/11324/commits/581d45615b0733a482044bf79caec4c43e21be83